tests/test_fits.py

```diff
@@ -79,5 +80,31 @@ def test_fits_reading():
     im = imageio.imread(compressed)
     assert im.shape == (2042, 3054)
 
+@pytest.mark.skipif("astropy is None")
+def test_fits_get_reader():
+    """Test reading fits with get_reader method"""
+    
+    #Set precedence as normal
+    teardown_module()
+    
+    if IS_PYPY:
+        return  # no support for fits format :(
+
+    import numpy as np
+    from astropy.io import fits
+    sigma = 10
+    xx, yy = np.meshgrid(np.arange(512),np.arange(512))
+    z = (1/(2*np.pi*(sigma**2)))*np.exp(-((xx**2)+(yy**2))/(2*(sigma**2)))
+    img = np.log(z)
+    phdu = fits.PrimaryHDU()
+    ihdu = fits.ImageHDU(img)
+    hdul = fits.HDUList([phdu,ihdu])
+    hdul.writeto('test.fits')
+    try:
+        im = imageio.get_reader('test.fits')
+        os.remove('test.fits')
+    except ValueError:
+        os.remove('test.fits')
+        pytest.fail('imageio.get_reader failed to find FITS loader')
```

Comment:
```suggestion
    im = imageio.get_reader('test.fits')
```

We can refactor the `os.remove` call, because we can use tmp_path (it is cleaner).

If we couldn't refactor `os.remove`, it would be cleaner to write

```python
try:
    pass # code here
except ValueError:
    pass # handle exception
finally:
    # will _always_ be executed, regardless of success or failure
    os.remove(...)
```

If a test produces an unhandled exception, pytest will mark the test as failed automatically, hence we do not have to specifically tell pytest to fail manually.

Reply:
Thanks for that suggestion! Part of the reason I ended up adding in the explicit fail was I hadn't realized the default format was set for `FITS` up earlier in the test file, so I was not testing in the right precedence scenario and was not getting a failure when I expected. Eventually saw the `teardown_module` function and was able to use it. But didn't refactor.